### PR TITLE
[WIP] Test GH actions for build

### PR DESCRIPTION
**Pull request by slide**

### CODE REVIEW

**Comment by slozier** [comment]
@slide I think this needs to be under `.github/workflows` to be picked up.

File: main.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+        submodules: true
+    - uses: actions/setup-dotnet@master
+      with:
+        dotnet-version: '3.0.100-preview8-013656'
```
Comment (slozier) [inline main.yml]: master is still on preview5

**Comment by slide** [comment]
Weird, I used the action editor thing to create the file.

**Comment by curtcurt871** [comment]
From 8051c6ea16d12d495fb00bad0a9c89f95935452e Mon Sep 17 00:00:00 2001
From: Alex Earl <slide.o.mix@gmail.com>
Date: Wed, 18 Sep 2019 16:23:52 -0700
Subject: [PATCH] Test GH actions for build

---
 main.yml | 82 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 82 insertions(+)
 create mode 100644 main.yml

diff --git a/main.yml b/main.yml
new file mode 100644
index 000000000..a8f434362
--- /dev/null
+++ b/main.yml
@@ -0,0 +1,82 @@
+name: IronPython2
+
+on: 
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  build:
+    name: Build/Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+        submodules: true
+    - uses: actions/setup-dotnet@master
+      with:
+        dotnet-version: '3.0.100-preview8-013656'
+    
+    - name: Grab Package Version
+      run: |
+        $xml = [xml] (Get-Content CurrentVersion.props)
+        $major = $xml.Project.PropertyGroup.MajorVersion
+        $minor = $xml.Project.PropertyGroup.MinorVersion
+        $micro = $xml.Project.PropertyGroup.MicroVersion
+        $serial = $xml.Project.PropertyGroup.ReleaseSerial
+        $level = $xml.Project.PropertyGroup.ReleaseLevel
+        if($level -eq 'final') {
+         $PackageVersion = "$major.$minor.$micro"
+        } elseif($level -ne 'final' -or $serial -ne '0') {
+         $PackageVersion = "$major.$minor.$micro-$level$serial"
+        }
+        # store the package version to an environment variable
+        Set-Content -Path "PackageVersion.txt" -Value $PackageVersion
+      shell: pwsh
+    
+    - name: Linux Version Information
+      if: startsWith(${{ matrix.os }}, "ubuntu")
+      run: |
+          # Testing and packaging tools
+          sudo apt-get -yq install mono-vbnc dos2unix fakeroot
+          # Dump some info about the tools
+          mono --version
+          msbuild /version
+          dotnet --info
+          df -Th
+          
+    - name: macOS Version Information
+      if: startsWith(${{ matrix.os }}, "macOS")
+      run: |
+          # use Mono 5.16.1 version
+          SYMLINK=5.16.1
+          MONOPREFIX=/Library/Frameworks/Mono.framework/Versions/$SYMLINK
+          echo "##vso[task.setvariable variable=DYLD_FALLBACK_LIBRARY_PATH;]$MONOPREFIX/lib:/lib:/usr/lib:$DYLD_LIBRARY_FALLBACK_PATH"
+          echo "##vso[task.setvariable variable=PKG_CONFIG_PATH;]$MONOPREFIX/lib/pkgconfig:$MONOPREFIX/share/pkgconfig:$PKG_CONFIG_PATH"
+          echo "##vso[task.setvariable variable=PATH;]$MONOPREFIX/bin:$PATH"
+          # Dump some info about the tools
+          mono --version
+          msbuild /version
+          dotnet --info
+          
+    - name: Build
+      run: pwsh make.ps1
+      
+    - name: Test
+      run: pwsh make.ps1 test-all
+      
+    - name: Test 32-bit
+      if: startsWith(${{ matrix.os }}, "windows")
+      run: pwsh make.ps1 -platform x86 test-all
+
+    - name: Package
+      if: succeededOrFailed()
+      run: pswh make.ps1 package

**Comment by curtcurt871** [comment]
From 8051c6ea16d12d495fb00bad0a9c89f95935452e Mon Sep 17 00:00:00 2001
From: Alex Earl <slide.o.mix@gmail.com>
Date: Wed, 18 Sep 2019 16:23:52 -0700
Subject: [PATCH] Test GH actions for build

---
 main.yml | 82 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 82 insertions(+)
 create mode 100644 main.yml

diff --git a/main.yml b/main.yml
new file mode 100644
index 000000000..a8f434362
--- /dev/null
+++ b/main.yml
@@ -0,0 +1,82 @@
+name: IronPython2
+
+on: 
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  build:
+    name: Build/Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+    
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+        submodules: true
+    - uses: actions/setup-dotnet@master
+      with:
+        dotnet-version: '3.0.100-preview8-013656'
+    
+    - name: Grab Package Version
+      run: |
+        $xml = [xml] (Get-Content CurrentVersion.props)
+        $major = $xml.Project.PropertyGroup.MajorVersion
+        $minor = $xml.Project.PropertyGroup.MinorVersion
+        $micro = $xml.Project.PropertyGroup.MicroVersion
+        $serial = $xml.Project.PropertyGroup.ReleaseSerial
+        $level = $xml.Project.PropertyGroup.ReleaseLevel
+        if($level -eq 'final') {
+         $PackageVersion = "$major.$minor.$micro"
+        } elseif($level -ne 'final' -or $serial -ne '0') {
+         $PackageVersion = "$major.$minor.$micro-$level$serial"
+        }
+        # store the package version to an environment variable
+        Set-Content -Path "PackageVersion.txt" -Value $PackageVersion
+      shell: pwsh
+    
+    - name: Linux Version Information
+      if: startsWith(${{ matrix.os }}, "ubuntu")
+      run: |
+          # Testing and packaging tools
+          sudo apt-get -yq install mono-vbnc dos2unix fakeroot
+          # Dump some info about the tools
+          mono --version
+          msbuild /version
+          dotnet --info
+          df -Th
+          
+    - name: macOS Version Information
+      if: startsWith(${{ matrix.os }}, "macOS")
+      run: |
+          # use Mono 5.16.1 version
+          SYMLINK=5.16.1
+          MONOPREFIX=/Library/Frameworks/Mono.framework/Versions/$SYMLINK
+          echo "##vso[task.setvariable variable=DYLD_FALLBACK_LIBRARY_PATH;]$MONOPREFIX/lib:/lib:/usr/lib:$DYLD_LIBRARY_FALLBACK_PATH"
+          echo "##vso[task.setvariable variable=PKG_CONFIG_PATH;]$MONOPREFIX/lib/pkgconfig:$MONOPREFIX/share/pkgconfig:$PKG_CONFIG_PATH"
+          echo "##vso[task.setvariable variable=PATH;]$MONOPREFIX/bin:$PATH"
+          # Dump some info about the tools
+          mono --version
+          msbuild /version
+          dotnet --info
+          
+    - name: Build
+      run: pwsh make.ps1
+      
+    - name: Test
+      run: pwsh make.ps1 test-all
+      
+    - name: Test 32-bit
+      if: startsWith(${{ matrix.os }}, "windows")
+      run: pwsh make.ps1 -platform x86 test-all
+
+    - name: Package
+      if: succeededOrFailed()
+      run: pswh make.ps1 package

**Pull request closed without merging**